statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,315 @@
+    corresponding to a 1D curve. This matrix is then decomposed using Principal
+    Components Analysis (PCA). This allows to represent the data using a finite
+    number of modes, or components. This compression process allows to turn the
+    functional representation into a scalar representation of the matrix. In
+    other words, you can visualize each curve from its components. With 2
+    components, this is called a bivariate plot.
+
+    This visualization exhibit a cluster of points. It indicate that a lot of
+    curve lead to common components. The center of the cluster is the mediane
+    curve. An the more you get away from the cluster, the more the curve is
+    unlikely to be similar to the other curves.
+
+    Using a kernel smoothing technique, the probability density function (PDF)
+    of the multivariate space can be recover. From this PDF, it is possible to
+    compute the density probability linked to the cluster and plot its
+    contours.
+
+    Finally, using these contours, the different quantiles can be extracted
+    allong with the median curve and the outliers.
+
+    Parameters
+    ----------
+    data : sequence of ndarrays or 2-D ndarray
+        The vectors of functions to create a functional boxplot from.  If a
+        sequence of 1-D arrays, these should all be the same size.
+        The first axis is the function index, the second axis the one along
+        which the function is defined.  So ``data[0, :]`` is the first
+        functional curve.
+    ncomp : int, optional
+        Number of components to use.  If None, returns the as many as the
+        smaller of the number of rows or columns in data.
+    alpha : list, optional
+        Extra percentile values to compute.
+    threshold : float
+        Percentile threshold value for outliers detection. High value means
+        a lower sensitivity to outliers. Default is `0.95`.
+    optimize: bool
+        Bandwidth optimization with cross validation or normal inferance
+    n_contours : int
+        Discretization per dimension of the reduced space.
+    xdata : ndarray, optional
+        The independent variable for the data.  If not given, it is assumed to
+        be an array of integers 0..N with N the length of the vectors in
+        `data`.
+    labels : sequence of scalar or str, optional
+        The labels or identifiers of the curves in `data`.  If given, outliers
+        are labeled in the plot.
+    ax : Matplotlib AxesSubplot instance, optional
+        If given, this subplot is used to plot in instead of a new figure being
+        created.
+
+    Returns
+    -------
+    fig : Matplotlib figure instance
+        If `ax` is None, the created figure.  Otherwise the figure to which
+        `ax` is connected.
+    hdr_res : dict
+
+         - 'median', array. Median curve.
+         - 'mean_percentile', array. 50% percentile band. [sup, inf] curves
+         - 'extreme_percentile', list of array. 90% percentile band. [sup, inf]
```

Comment:
I changed all percentile to quantile.

Regarding the bands yes it can be seen as a level of confidence. It is like a box plot with its 1.5 Inter Quartiles Ranges. I checked the original paper, they use: `50% HDR`. So I suggest:
`50_hdr` or `05_hdr` and for the label `50% HDR`.